### PR TITLE
Refactor: XML Policy to support user tracking and Json-like structure

### PR DESCRIPTION
**Pull request by keyur9**
Refactor: XML Policy to support user tracking and Json-like structure

# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 201-api-management-logs-to-moesif-using-eventhub-webapp/nested/microsoft.apimanagement/service/apis/policies.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
...
...n  </backend>\r\n  <outbound>\r\n    <log-to-eventhub logger-id=\"', parameters('logToEventhubLoggerName'), '\" partition-id=\"1\">\r\n      @{\r\n          var statusLine = string.Format(\"HTTP/1.1 {0} {1}\\r\\n\",\r\n                                              context.Response.StatusCode,\r\n                                              context.Response.StatusReason);\r\n\r\n          var body = context.Response.Body?.As&lt;string&gt;(true);\r\n          if (body != null &amp;&amp; body.Length &gt; 1024)\r\n          {\r\n              body = body.Substring(0, 1024);\r\n          }\r\n\r\n          var headers = context.Response.Headers\r\n                                          .Select(h =&gt; string.Format(\"{0}: {1}\", h.Key, String.Join(\", \", h.Value)))\r\n                                          .ToArray&lt;string&gt;();\r\n\r\n          var headerString = (headers.Any()) ? string.Join(\"\\r\\n\", headers) + \"\\r\\n\" : string.Empty;\r\n\r\n          return \"response:\"  + context.Variables[\"message-id\"] + \"\\n\"\r\n                              + statusLine + headerString + \"\\r\\n\" + body;\r\n     }\r\n  </log-to-eventhub>\r\n  </outbound>\r\n</policies>')]",
```
Comment (CodeYouMust) [inline 201-api-management-logs-to-moesif-using-eventhub-webapp/nested/microsoft.apimanagement/service/apis/policies.json:25]: You might have missed incorporating `parameters('logToEventhubLoggerName')` parameter and hardcoded it after export. Please set it as variable.

Did you deploy this ARM template along with other templates? Let me know if I can help you with it.

**Comment by CodeYouMust** [comment]
Verified that the new `ApimEventProcessor` builds + deploys using `201-api-management-logs-to-moesif-using-eventhub-webapp/scripts`?

**Comment by dkm199** [comment]
@keyur9  work with @CodeYouMust to get this merged.

File: 201-api-management-logs-to-moesif-using-eventhub-webapp/nested/microsoft.apimanagement/service/apis/policies.json (hunk 1 above)
Comment (dgilling) [reply 201-api-management-logs-to-moesif-using-eventhub-webapp/nested/microsoft.apimanagement/service/apis/policies.json:25]: @keyur9 is this fixed?

Comment (keyur9) [reply 201-api-management-logs-to-moesif-using-eventhub-webapp/nested/microsoft.apimanagement/service/apis/policies.json:25]: Yes, we've fixed it and checked in.

**Review by dgilling — APPROVED** [review]

**Pull request merged**